cli/src/main/java/org/projectnessie/tools/catalog/migration/cli/CatalogMigrationCLI.java

```diff
@@ -0,0 +1,507 @@
+      return 0;
+    }
+
+    printWriter.println(String.format("%nStarted %s ...", operation));
+
+    List<List<TableIdentifier>> IdentifierBatches = Lists.partition(identifiers, BATCH_SIZE);
+    AtomicInteger counter = new AtomicInteger();
+    IdentifierBatches.forEach(
+        identifierBatch -> {
+          catalogMigrator.registerTables(identifierBatch, resultBuilder);
+          printWriter.println(
+              String.format(
+                  "%nAttempted %s for %d tables out of %d tables.",
+                  operation, counter.incrementAndGet() * BATCH_SIZE, identifiers.size()));
+        });
+
+    CatalogMigrationResult result = resultBuilder.build();
+    writeToFile(
+        pathWithOutputDir(FAILED_IDENTIFIERS_FILE), result.failedToRegisterTableIdentifiers());
+    writeToFile(
+        pathWithOutputDir(FAILED_TO_DELETE_AT_SOURCE_FILE),
+        result.failedToDeleteTableIdentifiers());
+
+    printWriter.println(String.format("%nFinished %s ...", operation));
+    printSummary(result, printWriter, sourceCatalog.name(), targetCatalog.name());
+    printDetails(result, printWriter);
+    return 0;
+  }
+
+  private boolean canProceed(PrintWriter printWriter, Catalog sourceCatalog) {
+    if (isDryRun || disablePrompts) {
+      return true;
+    }
+    if (deleteSourceCatalogTables) {
+      if (sourceCatalog instanceof HadoopCatalog) {
+        printWriter.println(
+            String.format(
+                "[WARNING]: Source catalog type is HADOOP and it doesn't support dropping tables just from "
+                    + "catalog. %nAvoid operating the migrated tables from the source catalog after migration. "
+                    + "Use the tables from target catalog."));
+      }
+      return PromptUtil.proceedForMigration(printWriter);
+    } else {
+      return PromptUtil.proceedForRegistration(printWriter);
+    }
+  }
+
+  private List<TableIdentifier> processIdentifiersInput(PrintWriter printWriter) {
+    List<TableIdentifier> tableIdentifiers;
+    if (identifiersFromFile != null) {
+      try {
+        printWriter.println(
+            String.format("Collecting identifiers from the file %s...", identifiersFromFile));
+        printWriter.println();
+        tableIdentifiers =
+            Files.readAllLines(Paths.get(identifiersFromFile)).stream()
+                .map(TableIdentifier::parse)
+                .collect(Collectors.toList());
+      } catch (IOException e) {
+        throw new RuntimeException("Failed to read the file:", e);
+      }
+    } else if (!identifiers.isEmpty()) {
+      tableIdentifiers =
+          identifiers.stream().map(TableIdentifier::parse).collect(Collectors.toList());
+    } else {
+      tableIdentifiers = Collections.emptyList();
+    }
+    return tableIdentifiers;
+  }
+
+  private void validateIdentifierOptions() {
+    if (identifiersFromFile != null && !identifiers.isEmpty() && identifiersRegEx != null) {
+      throw new IllegalArgumentException(
```

Comment:
preconditions will have to check the same conditions multiple times as they cannot be nested (or I think so). Hence, I didn't use it initially for this.

I can change now.
It will look something like this 
```
Preconditions.checkArgument(
        !identifiers.isEmpty() && identifiersFromFile != null && identifiersRegEx != null,
        "All the three identifier options (`--identifiers`, `--identifiers-from-file`, "
            + "`--identifiers-regex`) are configured. Please use only one of them.");
    Preconditions.checkArgument(
        !identifiers.isEmpty() && identifiersFromFile != null,
        "Both `--identifiers` and "
            + "`--identifiers-from-file` options are configured. Please use only one of them.");
    Preconditions.checkArgument(
        identifiersFromFile != null && identifiersRegEx != null,
        "Both `--identifiers-from-file` and `--identifiers-regex` options are configured. Please use only one of them.");
    Preconditions.checkArgument(
        identifiersRegEx != null && !identifiers.isEmpty(),
        "Both `--identifiers-regex` and `--identifiers` options are configured. Please use only one of them.");
    Preconditions.checkArgument(
        identifiersFromFile != null && !Files.exists(Paths.get(identifiersFromFile)),
        "File specified in `--identifiers-from-file` option does not exist.");
```

Reply:
a) preconditions will have to check the same conditions multiple times as they cannot be nested (or I think so). Hence, I didn't use it initially for this.

b) The error message has to map for the negated result. So, it looks hard to read with multiple conditions.

If I change it, It will look something like this. Which I think is not easy to understand. 

    ```
Preconditions.checkArgument(
        !(!identifiers.isEmpty() && identifiersFromFile != null && identifiersRegEx != null),
        "All the three identifier options (`--identifiers`, `--identifiers-from-file`, "
            + "`--identifiers-regex`) are configured. Please use only one of them.");
    Preconditions.checkArgument(
        !(!identifiers.isEmpty() && identifiersFromFile != null),
        "Both `--identifiers` and "
            + "`--identifiers-from-file` options are configured. Please use only one of them.");
    Preconditions.checkArgument(
        !(identifiersFromFile != null && identifiersRegEx != null),
        "Both `--identifiers-from-file` and `--identifiers-regex` options are configured. Please use only one of them.");
    Preconditions.checkArgument(
        !(identifiersRegEx != null && !identifiers.isEmpty()),
        "Both `--identifiers-regex` and `--identifiers` options are configured. Please use only one of them.");
    Preconditions.checkArgument(
        !(identifiersFromFile != null && !Files.exists(Paths.get(identifiersFromFile))),
        "File specified in `--identifiers-from-file` option does not exist.");
```